### PR TITLE
Upgrade tokio-util to 0.7

### PR DESCRIPTION
**Pull request by djc**

### CODE REVIEW

**Comment by olix0r** [comment]
I'm not sure why the nightly build failed but it looks like it's picking up a very old tokio version:
```
Compiling tokio v1.0.1
```

**Comment by djc** [comment]
That's because it's the `cargo update -Zminimal-versions` step of the CI build. I already asked on the Tokio Discord about this, we should likely get a new tokio-util that fixes the minimal version of Tokio that it needs.

**Comment by djc** [comment]
I submitted https://github.com/tokio-rs/tokio/pull/4521 to try and get a tokio-util 0.7.1 out.

File: Cargo.toml [hunk 1]
```diff
@@ -43,7 +43,7 @@ members = [
 futures-core = { version = "0.3", default-features = false }
 futures-sink = { version = "0.3", default-features = false }
 futures-util = { version = "0.3", default-features = false }
-tokio-util = { version = "0.6", features = ["codec"] }
+tokio-util = { version = "0.7", features = ["codec"] }
```
Comment (paolobarbolini) [inline Cargo.toml:46]: tokio-util 0.7.1 is out!

```suggestion
tokio-util = { version = "0.7.1", features = ["codec"] }
```

**Comment by djc** [comment]
@paolobarbolini thanks for the reminder! I think this could be merged now.

**Comment by paolobarbolini** [comment]
Could we get a new release with this @seanmonstar? I'd love not having to live with duplicate dependency versions in my dependency tree

**Comment by seanmonstar** [comment]
Published: https://github.com/hyperium/h2/releases/tag/v0.3.13

**Pull request merged**